### DISCUSSION TITLE
How do you exclude aspects with io.freefair.aspectj.post-compile-weaving?

### DISCUSSION THREAD

**Discussion by libertywork-agb** [freefair/gradle-plugins#471] [Q&A] [answered]
I have a Gradle 7.3.3 aspect library and a Gradle 7.3.3 project which consumes that library. I am applying the `post-compile-weaving` plugin in the project.
```groovy
plugins {
    id 'io.freefair.aspectj.post-compile-weaving' version '6.3.0'
}
```

I am then adding the library to the project as an aspect dependency.
```groovy
dependencies {
    aspect 'com.example:my-aspect-lib:0.0.1'
}
```

Aside from the library aspects that I want to weave post-compile, the project itself has aspects that I want Spring AOP to handle at runtime. Those aspects must not be woven post-compile. However, the default behavior of `post-compile-weaving` is to weave all aspects in the project.

I have tried quite a few different iterations of ajc configuration, including messing with `compilerArgs`, but so far I have not been able to exclude the project aspects from post-compile weaving.
```groovy
compileJava {
    ajc {
        options {
            aspectpath.minus files()
            aspectpath.setFrom configurations.aspect
        }
    }
}
```

Is it possible to exclude aspects with the `post-compile-weaving` plugin?

**Comment by larsgrefer** [comment, marked as answer]
I don't think it's possible to exclude a specific class from being processed, because the whole compiler output directory (`build/classes/java/main`) is passed to ajc as `-inpath`.

You could try to move your _Spring AOP_ aspect to a different subproject, which does not have the post-compile-weaving applied
